### PR TITLE
Add ExcludeTimeFilter

### PR DESCRIPTION
**Pull request by sugolov**

### CODE REVIEW

File: src/main/java/filters/ExcludeTimeFilter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import entities.Schedule;
+import entities.Section;
+import entities.Timeslot;
+
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
```
Comment (kenneth-miura) [inline src/main/java/filters/ExcludeTimeFilter.java:11]: This description makes sense with the old way Filters worked, but now it doesn't. I like the general idea tho

File: src/main/java/filters/ExcludeTimeFilter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package filters;
+
+import entities.Schedule;
+import entities.Section;
+import entities.Timeslot;
+
+import java.sql.Time;
```
Comment (kenneth-miura) [inline src/main/java/filters/ExcludeTimeFilter.java:7]: I don't think this import is needed as far as I can tell

File: src/main/java/filters/ExcludeTimeFilter.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        for (Section lec : s.getLectures()) {
+            timeslots.addAll(lec.getTimes());
+        }
+
+        for (Section tut : s.getTutorials()) {
+            timeslots.addAll(tut.getTimes());
+        }
+
+        for (Timeslot exclude : timeExclusions) {
```
Comment (kenneth-miura) [inline src/main/java/filters/ExcludeTimeFilter.java:52]: I think if u sort the timeslots, you only need to iterate through once? And just check the conflicts with the timeslot that's right next to it

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
looks good generally, there's a few points

File: src/main/java/filters/ExcludeTimeFilter.java (hunk 3 above)
Comment (roryhofland) [reply src/main/java/filters/ExcludeTimeFilter.java:52]: Yeah, sorting the excluded timeslots would mean that all could be checked in one iteration. When you reach an excluded timeslot and it passes, just resume checking for the next one.

Comment (sugolov) [reply src/main/java/filters/ExcludeTimeFilter.java:52]: Yea u right

**Comment by sugolov** [comment]
we should merge the PR making timeslot comparable before i finalize this

**Comment by sugolov** [comment]
actually maybe we should implement everything using sorted lists in another PR as multiple filters will change

**Pull request closed without merging**